File: lang/nodejs/src/vac.fun/util/SequenceBuilder.js
"use strict"

const RandomFactory = require("vac.fun/util/RandomFactory")
const Sequences     = require("vac.fun/util/Sequences")

exports.packIncreasing = (arr) => {
    RandomFactory.seed()
    arr[0] = RandomFactory.integerN(1, 3)
    for (let i = 1, length = arr.length; i < length; i++) {
        arr[i] = arr[i - 1] + RandomFactory.integerN(1, 3)
    }
}

exports.packRandom = (arr) => {
    RandomFactory.seed()
    for (let i = 0, length = arr.length; i < length; i++) {
        arr[i] = RandomFactory.generateInteger()
    }
}

exports.packDecreasing = (arr) => {
    this.packIncreasing(arr)
    Sequences.reverse(arr)
}

File: rust/muse/src/util/sequence_builder.rs
use super::{random_factory, sequences};

pub fn pack_identical(arr: &mut [i32]) {
    let n = random_factory::gen_int();
    arr.fill(n);
}

pub fn pack_increasing(arr: &mut [i32]) {
    if arr.is_empty() {
        return;
    }
    arr[0] = random_factory::gen_int_n(1, 3);
    (1..arr.len()).for_each(|i| {
        arr[i] = arr[i - 1] + random_factory::gen_int_n(1, 3);
    });
}

pub fn pack_random(arr: &mut [i32]) {
    arr.fill_with(random_factory::gen_int)
}

pub fn pack_decreasing(arr: &mut [i32]) {
    pack_increasing(arr);
    sequences::reverse(arr);
}
